src/morphodict/lexicon/management/commands/importjsondict.py

```diff
@@ -0,0 +1,174 @@
+import json
+import logging
+from argparse import ArgumentParser, BooleanOptionalAction
+from pathlib import Path
+
+from django.conf import settings
+from django.core.management import BaseCommand, call_command
+from django.db import transaction
+from tqdm import tqdm
+
+from CreeDictionary.CreeDictionary.paradigm.generation import default_paradigm_manager
+from CreeDictionary.utils.english_keyword_extraction import stem_keywords
+from morphodict.analysis import RichAnalysis, strict_generator
+from morphodict.lexicon.models import (
+    Wordform,
+    Definition,
+    DictionarySource,
+    TargetLanguageKeyword,
+    SourceLanguageKeyword,
+)
+from morphodict.lexicon.util import to_source_language_keyword
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser: ArgumentParser):
+        parser.add_argument(
+            "--purge",
+            action=BooleanOptionalAction,
+            # TODO: change default to False once working with full DBs
+            default=True,
+            help="""
+                Delete all existing entries that are not found in this import
+                file. Used for importing a full dictionary file when keys may
+                have changed.
+            """,
+        )
+        parser.add_argument(
+            "--atomic",
+            action=BooleanOptionalAction,
+            default=settings.DEBUG,
+            help="""
+                Run the import in a single transaction. This will make the
+                import run much faster, but will lock up the database so that
+                other processes can’t access it. Good for development use.
+            """,
+        )
+        parser.add_argument("json_file")
+
+    def handle(self, json_file, purge, atomic, **options):
+        if atomic:
+            with transaction.atomic():
+                self.run_import(json_file=json_file, purge=purge)
+        else:
+            self.run_import(json_file=json_file, purge=purge)
+
+    def run_import(self, json_file, purge):
+        for abbrv in ["CW", "MD", "AE", "ALD", "OS"]:
+            if not DictionarySource.objects.filter(abbrv=abbrv):
+                DictionarySource.objects.create(abbrv=abbrv)
```

Comment:
So every site will have dictionary source entries for all other sites? This seems... odd.

I think DictionarySources will need to be addressed in a future PR — there's definitely some work to be done here, but it's definitely okay to punt that work until later!